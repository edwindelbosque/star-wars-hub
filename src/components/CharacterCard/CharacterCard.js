import React from 'react';
import './CharacterCard.scss';
import PropTypes from 'prop-types';

const CharacterCard = ({ name, species, homeworldName, homeworldPop, relatedFilms, addFavorite, removeFavorite }) => {
    const filmsList = relatedFilms.map(film => {
        return (
                <li className='character-movie-list'>{film}</li>
        )
    })
    return (
        <article className='character-info'>
            <h5 className='character-name'>{name}</h5>
            <hr />
            <h5 className='character-species'>species: <span>{species}</span></h5>
            <h5 className='character-homeworld'>
                homeworld: <span>{homeworldName}</span>
                <br /> 
                population: <span>{homeworldPop}</span> </h5>
            <h5 className='character-rel-movies'>appears in: {filmsList}</h5>
            <div className='character-footer' onClick={() => addFavorite({ name, species, homeworldName, homeworldPop, relatedFilms})}>
                Add {name} to favorites 
            </div>
            <div className='footer-remove' onClick={() => removeFavorite({ name, species, homeworldName, homeworldPop, relatedFilms})}>
              Remove {name} from favorites
            </div>
        </article>
    )
}

CharacterCard.propTypes = {
  name: PropTypes.string,
  species: PropTypes.array,
  homeworldName: PropTypes.string,
  homeworldPop: PropTypes.string,
  relatedFilms: PropTypes.array,
};

export default CharacterCard;