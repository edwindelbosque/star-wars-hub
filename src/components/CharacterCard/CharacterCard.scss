.character-info {
    border-radius: 4px;
    width: 70%;
    color: white;
    font-size: 20px;
    margin: auto;
    margin-bottom: 20px;
    margin-top: 10px;
    background: linear-gradient(to bottom right, #435a7d9e, #38383800);
    border: 1px solid rgb(186, 186, 186);
    border-bottom: 1px solid rgba(169, 82, 255, 0.639);
    border-top: 1px solid rgba(93, 239, 255, 0.733);
    border-left: 1px solid rgba(93, 150, 255, 0.616);
    border-right: 1px solid rgba(93, 163, 255, 0.687);
    box-shadow: 0 0 30px #6797ff4c;
}

.character-name {
    font-size: 2.3em;
    color: rgb(255, 234, 0);
    padding: 10px 30px;
    text-align: left;
    font-weight: 900;
}

.character-species,
.character-homeworld {
    font-size: 1em;
    text-align: initial;
    padding: 8px 30px;
    font-weight: 200;
}

.character-homeworld {
    font-size: 1.3em;
    text-align: initial;
    padding: 8px;
}

.character-rel-movies {
    font-size: 1em;
    text-align: justify;
    padding: 10px 30px;
    font-weight: 200;
    padding-bottom: 30px;
}

.character-movie-list {
    list-style: none;
    font-weight: 400;
    margin-top: 5px;
    color: rgba(184, 252, 167, 0.776);
}

.character-footer,
.footer-remove {
    margin: 25px;
    text-align: end;
}

span {
    color: #98e7ff;
    text-transform: uppercase;
    font-size: 1.3em;
    font-weight: 600;
}

li {
    text-transform: uppercase;
    font-weight: 200;
}

.button-favorite {
    padding: 5px;
    background-color: rgb(255, 225, 0);
    color: black;
    border: 1px solid rgb(192, 174, 36);
    border-radius: 4px;
    cursor: pointer;
    font-weight: 600;
    transition: .3s;
    margin-left: 285px;
    margin-bottom: 30px;

    &:hover {
        box-shadow: 0 0 8px yellow;
        transition: .2s;
        background-color: rgb(255, 234, 117);
    }

    &:active {
        box-shadow: 0 0 15px yellow;
        background-color: white;
        transition: .2s;
    }

    &:focus {
        outline: none;
    }
}