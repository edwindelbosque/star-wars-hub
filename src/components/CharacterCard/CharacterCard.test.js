import React from 'react';
import { shallow } from 'enzyme';
import CharacterCard from './CharacterCard';

describe('CharacterCard', () => {
  it('component should match the snapshot', () => {
    const wrapper = shallow(
      <CharacterCard
        key={3}
        name={'Darth Vader'}
        species={['Human']}
        homeworldName={'Death Star'}
        homeworldPop={'3'}
        relatedFilms={[]}
      />
    )
    expect(wrapper).toMatchSnapshot();
  });
})