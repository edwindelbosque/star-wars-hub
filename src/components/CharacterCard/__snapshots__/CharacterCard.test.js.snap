// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard component should match the snapshot 1`] = `
<article
  className="character-info"
>
  <h5
    className="character-name"
  >
    Darth Vader
  </h5>
  <hr />
  <h5
    className="character-species"
  >
    species: 
    <span>
      Human
    </span>
  </h5>
  <h5
    className="character-species"
  >
     homeworld: 
    <span>
      Death Star
    </span>
  </h5>
  <h5
    className="character-species"
  >
    population: 
    <span>
      3
    </span>
     
  </h5>
  <h5
    className="character-rel-movies"
  >
    appears in: 
  </h5>
  <button
    className="button-favorite"
    onClick={[Function]}
  >
    Add to favorites
  </button>
</article>
`;
